internal/pkg/consensus/validation.go

```diff
@@ -118,25 +130,66 @@ func NewMessageSyntaxValidator() *MessageSyntaxValidator {
 	return &MessageSyntaxValidator{}
 }
 
+// Validates message syntax and state-independent invariants.
 func (v *MessageSyntaxValidator) Validate(ctx context.Context, smsg *types.SignedMessage) error {
-	// check non-state dependent invariants
-	msg := smsg.Message
-	if msg.GasLimit > types.BlockGasLimit {
-		log.Debugf("Message: %s gas limit from actor: %s above block limit: %s", msg, msg.From, types.BlockGasLimit)
-		errGasAboveBlockLimitCt.Inc(ctx, 1)
-		return errGasAboveBlockLimit
+	msg := &smsg.Message
+	var msgLen int
+	if smsg.Signature.Type == crypto.SigTypeBLS {
+		enc, err := smsg.Message.Marshal()
+		if err != nil {
+			return errors.Wrapf(err, "failed to calculate message size")
+		}
+		msgLen = len(enc)
+	} else {
+		enc, err := smsg.Marshal()
+		if err != nil {
+			return errors.Wrapf(err, "failed to calculate message size")
+		}
+		msgLen = len(enc)
 	}
 
-	if msg.Value.LessThan(specsbig.Zero()) {
-		log.Debugf("Cannot transfer negative value: %s from actor: %s", msg.Value, msg.From)
-		errNegativeValueCt.Inc(ctx, 1)
-		return errNegativeValue
+	if msg.To.Empty() {
+		invReceiverUndefCt.Inc(ctx, 1)
+		return fmt.Errorf("empty receiver: %s", msg)
 	}
-
-	if msg.GasPrice.LessThanEqual(types.ZeroAttoFIL) {
-		return errGasPriceZero
+	if msg.From.Empty() {
+		invSenderUndefCt.Inc(ctx, 1)
+		return fmt.Errorf("empty sender: %s", msg)
+	}
+	if msg.CallSeqNum < 0 {
+		invCallSeqNegativeCt.Inc(ctx, 1)
+		return fmt.Errorf("negative callseqnum %d: %s", msg.CallSeqNum, msg)
+	}
+	if msg.Value.LessThan(big.Zero()) {
+		invNegativeValueCt.Inc(ctx, 1)
+		return fmt.Errorf("negative value %s: %s", msg.Value, msg)
+	}
+	if msg.Value.GreaterThan(msgMaxValue) {
+		invValueAboveMaxCt.Inc(ctx, 1)
+		return fmt.Errorf("value %s exceeds max %s: %s", msg.Value, msgMaxValue, msg)
 	}
+	// The spec calls for validating a non-negative method num, but by the time it's decoded into a uint64
+	// we can't do that any more.
```

Comment:
It's good to call this out here to help the reader see that this check is happening elsewhere.  I think the spec should express this opinion as a requirement that this field is serialized as an unsigned integer (major type 0) and never a negative integer (major type 1). I think the standard way to do this is by changing `CallSeqNum` [defined in this section](https://filecoin-project.github.io/specs/#systems__filecoin_vm__actor) to a specs "uint64".  Then this comment can be removed.

If we don't want to do this then in my view the current approach is still working as intended, its just an implementation detail that we cover this validation rule on the decode step.  The comment can change a bit to reflect this "we can't do that any more" => "we have already checked this"

Finally if there is some reason that we want to allow negative CallSeqNums that fail validation then we can change the field type to int64 and add a check here.